landoapi/api/try_push.py

```diff
@@ -53,38 +63,52 @@ def build_revision_from_patch_helper(helper: PatchHelper) -> Revision:
 
 def convert_json_patch_to_bytes(patch: str) -> bytes:
     """Convert from the base64 encoded patch to `bytes`."""
-    return base64.b64decode(patch.encode("ascii"))
+    try:
+        return base64.b64decode(patch.encode("ascii"))
+    except binascii.Error:
+        raise ProblemException(
+            400,
+            "Patch decoding error.",
+            "A patch could not be decoded from base64.",
+            type="https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400",
+        )
 
 
 def parse_revisions_from_request(
-    patches: list[str], patch_format: str
+    patches: list[str], patch_format: PatchFormat
 ) -> list[Revision]:
     """Convert a set of base64 encoded patches to `Revision` objects."""
     patches_bytes = (
         io.BytesIO(convert_json_patch_to_bytes(patch)) for patch in patches
     )
 
-    if patch_format == "hgexport":
-        return [
-            build_revision_from_patch_helper(HgPatchHelper(patch))
-            for patch in patches_bytes
-        ]
-
-    if patch_format == "git-format-patch":
-        return [
-            build_revision_from_patch_helper(GitPatchHelper(patch))
-            for patch in patches_bytes
-        ]
-
-    raise ValueError(f"Unknown value for `patch_format`: {patch_format}.")
+    try:
+        if patch_format == PatchFormat.HgExport:
+            return [
+                build_revision_from_patch_helper(HgPatchHelper(patch))
+                for patch in patches_bytes
+            ]
+
+        if patch_format == PatchFormat.GitFormatPatch:
+            return [
+                build_revision_from_patch_helper(GitPatchHelper(patch))
+                for patch in patches_bytes
+            ]
```

Comment:
- Ideally the try block will only wrap the specific calls that would cause an exception
- The two if statements should go outside the try block